backend/src/main/java/com/woowacourse/friendogly/club/service/ClubQueryService.java

```diff
@@ -18,23 +23,56 @@
 public class ClubQueryService {
 
     private final ClubRepository clubRepository;
+    private final MemberRepository memberRepository;
+    private final PetRepository petRepository;
 
-    public ClubQueryService(ClubRepository clubRepository) {
+    public ClubQueryService(
+            ClubRepository clubRepository,
+            MemberRepository memberRepository,
+            PetRepository petRepository
+    ) {
         this.clubRepository = clubRepository;
+        this.memberRepository = memberRepository;
+        this.petRepository = petRepository;
     }
 
-    public List<FindSearchingClubResponse> findSearching(FindSearchingClubRequest request) {
+    public List<FindSearchingClubResponse> findSearching(Long memberId, FindSearchingClubRequest request) {
+        Member member = memberRepository.getById(memberId);
+        List<Pet> pets = petRepository.findByMemberId(memberId);
+
         Specification<Club> spec = ClubSpecification.where()
                 .equalsAddress(request.address())
                 .hasGenders(request.genderParams())
                 .hasSizeTypes(request.sizeParams())
                 .build();
 
-        return clubRepository.findAll(spec).stream()
+        List<Club> clubs = clubRepository.findAll(spec);
+
+        if (FilterCondition.toFilterCondition(request.filterCondition()) == FilterCondition.ABLE_TO_JOIN) {
+            return clubs.stream()
+                    .filter(club -> club.isJoinable(member, pets))
+                    .map(club -> new FindSearchingClubResponse(club, collectOverviewPetImages(club)))
+                    .toList();
+        }
```

Comment:
```suggestion
        if (FilterCondition.from(request.filterCondition()) == ABLE_TO_JOIN) {
            return clubs.stream()
                    .filter(club -> club.isJoinable(member, pets))
                    .map(club -> new FindSearchingClubResponse(club, collectOverviewPetImages(club)))
                    .toList();
        }
```

filter condition이라는 말이 너무 많아서 헷갈리는데
메서드명 개선하고 static import 하면 어떨까요?

Reply:
동일성 비교를 메소드 내부로 보내면서 개선해봤습니다.